plugin/helpers_bots.go

```diff
@@ -157,6 +167,10 @@ func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, options ...ShouldPr
 		return false, nil
 	}
 
+	if post.Props["from_webhook"] == true && !messageProcessOptions.AllowWebhook {
```

Comment:
Could you please move this after the SystemMessage check? Given that we only need to check a field, it's very lightweight and can be done first.

Reply:
Maybe I was ambitious here. I meant moving this code piece to https://github.com/mattermost/mattermost-server/pull/13270/files#diff-7c191d59bfe26e51b1828e40ccfc46f8R150. Does it make my feedback more clear?